File: akka/src/main/java/com/github/uryyyyyyy/samples/akka/java/helloworld/pingpong/ChildActor.java
package com.github.uryyyyyyy.samples.akka.java.helloworld.pingpong;

import akka.actor.UntypedActor;

public class ChildActor extends UntypedActor {

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof String) {
			System.out.println("pong received: " + message);
			getSender().tell(message, getSender());
		} else {
			System.out.println("Got unknown type message.");
			unhandled(message);
		}
	}
}

File: akka/src/main/scala/com/github/uryyyyyyy/samples/akka/helloworld/pingpong/ChildActor.scala
package com.github.uryyyyyyy.samples.akka.helloworld.pingpong

import akka.actor.Actor
import MyMessage.ReturnMessage

class ChildActor extends Actor {
	def receive = {
		case i:Int => {
			println("ChildActor - calc 100 * 2")
			val res = ReturnMessage(i * 2)
			sender ! res
		}
	}
}